examples/player/simid_player.js

```diff
@@ -315,26 +320,45 @@ class SimidPlayer {
    */
   videoComplete() {
       this.simidProtocol.sendMessage(MediaMessage.ENDED);
-      // once an ad is complete an the iframe should be hidden
+
+      if (this.requestedDuration_ == NO_REQUESTED_DURATION) {
+        this.stopAd(StopCode.MEDIA_PLAYBACK_COMPLETE);
+      } else if (this.requestedDuration_ != UNLIMITED_DURATION) {
+        // The creative has requested a different completion duration, so use that duration.
+        const durationExtensionMs = (this.requestedDuration_ - this.adVideoElement_.duration) * 1000;
+        setTimeout(() => {
+          // The creative has suggested a different close time, so label this creative_initiated.
+          this.stopAd(StopCode.CREATIVE_INITATED);
+        }, durationExtensionMs);
+      }
```

Comment:
This only works if `requestedDuration` extends the `adVideoElement_.duration`. If you want to shorten the duration it will have no effect